### DISCUSSION TITLE
getProperty("facing") not working as expected

### DISCUSSION THREAD

**Discussion by JackM33** [Minestom/Minestom#1040] [Help]
Hi I'm trying to get the facing property of an existing block and I'm not sure why but whenever I call getProperty("facing") it throws an exception saying it returned null. As I understood it the string I pass it is the same name as the vanilla properties? Or am I doing this completely wrong? 

```
Block childBlock = event.getInstance().getBlock(event.getBlockPosition().relative(BlockFace.values()[i]));

if (childBlock.getProperty("facing").contains("west")) { // this could be any direction, I pass it a string defined earlier but I omitted that
    event.getInstance().breakBlock(event.getPlayer(), event.getBlockPosition().relative(BlockFace.values()[i]));
}
```

![image](https://user-images.githubusercontent.com/47346677/167272558-a1891a25-90f3-4dfd-af91-b07801d382a2.png)

Thanks!

**Comment by MrlnHi** [comment]
It should work just fine, I think. Try debugging if your childBlock is the one you expect it to be

**Reply by JackM33** [reply]
hmm the thing is it's returning null. Shouldn't it return the default north no matter what block? or do some blocks not have the property at all?

**Reply by MrlnHi** [reply]
is childBlock.getProperty("facing") returning null or is event.getInstance().getBlock?

**Reply by MrlnHi** [reply]
Maybe add the exception you get to the discussion

**Reply by JackM33** [reply]
definitely yeah sorry I forgot to do that.

**Reply by MrlnHi** [reply]
now that I saw the exception I'd say that it's probably because you are trying to get the facing value of a block which doesn't have that property

**Reply by JackM33** [reply]
ah ok thanks I'll look into it. I might've assumed a block I was testing had it.

**Reply by JackM33** [reply]
Oh ok so as I understand it vines have a totally different property that appears to do the same sort of thing as do torches and presumably a bunch of other stuff.

**Reply by MrlnHi** [reply]
Yes, as vines don't look north but can have a vine at north, south and west in the same block
